File: hw/vaca/u3/HWStrategy/src/main/java/ec/edu/espe/hwstrategy/SortingContext.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.hwstrategy;

/**
 *
 * @author Kevin Vaca Edison's OOP ESPE
 */
public class SortingContext {
       private SortingStrategy strategy;
    
    public int[] sort(int[] data) {
        int size = data.length;
        setSortStrategy(size);
        return strategy.sort(data);
    }
    
    private void setSortStrategy(int size) {
        if (size < 10) {
            this.strategy = new BubbleSort();
        } else if (size < 100) {
            this.strategy = new InsertionSort();
        } else {
            this.strategy = new QuickSort();
        }
        System.out.println("Strategy set to: " + strategy.getClass().getSimpleName());
    }
    
    // Optional: Allow manual strategy setting
    public void setStrategy(SortingStrategy strategy) {
        this.strategy = strategy;
    }
}

File: hw/fuentes/U3/HW22STRATEGY/SortAppPY/ec/edu/espe/clientApp/model/sorting_context.py
from model.bubble_sort import BubbleSort
from model.insertion_sort import InsertionSort
from model.quick_sort import QuickSort

class SortingContext:
    def __init__(self):
        self._strategy = None

    def sort(self, data: list[int]) -> list[int]:
        size = len(data)
        self._strategy = self._set_strategy(size)
        return self._strategy.sort(data)

    def _set_strategy(self, n: int):
        if 0 < n < 30:
            return BubbleSort()
        elif 30 <= n < 100:
            return InsertionSort()
        else:
            return QuickSort()
